pydocx/export/html.py

```diff
@@ -142,268 +152,711 @@ def style(self):
 
-    def paragraph(self, text, pre=None):
-        return self.make_element(
-            tag='p',
-            contents=text,
-        )
+    def export_footnote(self, footnote):
+        results = super(PyDocXHTMLExporter, self).export_footnote(footnote)
+        tag = HtmlTag('li')
+        return tag.apply(results, allow_empty=False)
 
-    def heading(self, text, heading_style_name):
-        heading_value = self.heading_level_conversion_map.get(
-            heading_style_name,
-            self.default_heading_level,
-        )
-        return self.make_element(
-            tag=heading_value,
-            contents=text,
-        )
+    def _is_ordered_list(self, numbering_level):
+        return not numbering_level.is_bullet_format()
 
-    def insertion(self, text, author, date):
-        return self.make_element(
-            tag='span',
-            contents=text,
-            attrs={
-                'class': 'pydocx-insert',
-            },
-        )
+    def get_numbering_tracking(self, paragraph):
+        numbering_tracking = self.numbering_tracking.get(paragraph.parent)
+        if not numbering_tracking:
+            return
 
-    def hyperlink(self, text, href):
-        if text == '':
-            return ''
-        return self.make_element(
-            tag='a',
-            contents=text,
-            attrs={
-                'href': href,
-            },
-        )
+        tracking = numbering_tracking.get(paragraph)
+        if not tracking:
+            return
 
-    def image_handler(self, image_data, filename, uri_is_external):
-        if uri_is_external:
-            return image_data
-        extension = filename.split('.')[-1].lower()
-        b64_encoded_src = 'data:image/%s;base64,%s' % (
-            extension,
-            base64.b64encode(image_data).decode(),
-        )
-        b64_encoded_src = self.escape(b64_encoded_src)
-        return b64_encoded_src
-
-    def image(self, image_data, filename, x, y, uri_is_external):
-        src = self.image_handler(image_data, filename, uri_is_external)
-        if not src:
-            return ''
-        if all([x, y]):
-            return '<img src="%s" height="%s" width="%s" />' % (
-                src,
-                y,
-                x,
+        return tracking
+
+    def get_numbering_level_tag_begin(self, paragraph):
+        tracking = self.get_numbering_tracking(paragraph)
+        if not tracking:
+            return
+
+        level = tracking.get('open-level')
+        if level is not None:
+            pydocx_class = 'pydocx-list-style-type-{fmt}'.format(
+                fmt=level.num_format,
             )
-        else:
-            return '<img src="%s" />' % src
-
-    def deletion(self, text, author, date):
-        return self.make_element(
-            tag='span',
-            contents=text,
-            attrs={
-                'class': 'pydocx-delete',
-            },
+            attrs = {}
+            if self._is_ordered_list(level):
```

Comment:
How about:

```python
if self._is_ordered_list(level):
    attrs['class'] = pydocx_class
    tag_name = 'ol'
else:
    tag_name = 'ul'
return HtmlTag(tag_name, **attrs)
```

I don't see any reason to return out of each nesting. Thoughts?